### PR TITLE
ENH: HelpfulIntEnum

### LINKED ISSUE #41: Add HelpfulIntEnum from pcdsdaq
https://github.com/pcdshub/pcdsdaq/pull/108/files#r785021109
https://github.com/pcdshub/pcdsdaq/pull/108/files#r785022116

pcdsdaq/pcdsdaq/daq/interface.py

### PR DESCRIPTION
**Pull request by klauer**
## Description
Adds HelpfulIntEnum with case-insensitive functionality

## Motivation and Context
* Closes #41 
* Marked @ZLLentz as original committer for helpfulintenum from pcdsdaq

## How Has This Been Tested?
* Small suite of tests here (I don't think there were any in pcdsdaq?)

## Where Has This Been Documented?
PR and issue

### CODE REVIEW

File: pcdsutils/enum.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        ----------
+        identifier : EnumId
+            Any str, int, or Enum value that corresponds with a valid value
+            on this HelpfulIntEnum instance.
+
+        Returns
+        -------
+        enum : HelpfulIntEnum
+            The corresponding enum object associated with the identifier.
+        """
```
Comment (ZLLentz) [inline pcdsutils/enum.py:47]: To document the `ValueError`:
```suggestion
        Raises
        ------
        ValueError
            If the identifier cannot be matched to any enum defined here.
        """
```
Based on https://numpydoc.readthedocs.io/en/latest/format.html#raises

**Comment by ZLLentz** [comment]
It's probably worth tagging after this since we want to use it in both pcdsdevices and pcdsdaq, right?

File: pcdsutils/enum.py (hunk 1 above)
Comment (klauer) [reply pcdsutils/enum.py:53]: Ack, sorry - I had something like that and forgot to push that commit. That description look OK? I don't mind switching it up to this version.

**Review by ZLLentz — APPROVED** [review]
Feels weird reviewing this one. Tests look solid, glad to lose the case sensitivity.

File: pcdsutils/enum.py (hunk 1 above)
Comment (ZLLentz) [reply pcdsutils/enum.py:53]: Yeah looks good

**Comment by klauer** [comment]
Yeah, a tag would be appropriate. If we get it out in time, we can also remove the vendored version in the other PR.

**Comment by ZLLentz** [comment]
One last thought- did we also want to have some handler on `_missing_`? I think this is distinct in that in runs on the case where we do `my_enum(4)` as well as on some set of the existing cases? I'm having trouble finding the details and also having trouble identifying a place where this saves us, but it is worth thinking about.

**Review by tangkong — APPROVED** [review]
Looks good to me, I like how the changes clean up the downstream code.

**Comment by klauer** [comment]
I'm not sure, @ZLLentz - I added some test cases there to hit it, at least. What would you expect it to do in the case of an int enum?

**Comment by ZLLentz** [comment]
I couldn't think of anything specific- it just seems like the right place to do "something" about it if we decided there was a thing.

**Pull request merged**